cli/src/alluxio.org/cli/cmd/generate/template.go

```diff
@@ -0,0 +1,62 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package generate
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	"alluxio.org/cli/env"
+	"alluxio.org/log"
+)
+
+var Template = &TemplateCommand{}
+
+type TemplateCommand struct {
+	conf string
+}
+
+func (c *TemplateCommand) ToCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "template --conf",
+		Short: "Generate a config file if one doesn't exist.",
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 {
+				log.Logger.Infoln("Usage: template <alluxio_master_hostname>")
+				return
+			}
+			targetDir := filepath.Join(env.ConfAlluxioConfDir.EnvVar, "alluxio-site.properties")
+			content := "alluxio.master.hostname=" + args[0]
+			if _, err := os.Stat(targetDir); err != nil {
```

Comment:
we can reorganize this logic a little bit to make it less nested and easier to understand
current:
```
if statError
	if not exists
		write file
		if cannot write
			error writing file and return
		log file written and return
	else
		error reading and return
else
	warn file exists and return
```

one less nested level and less return statements:
```
_, statErr := os.Stat(targetDir)
if file exists => os.IsExist(err)
	warn file exists
else if file not exists => os.IsNotExist(err)
	write file
	if cannot write
		error writing file
	log file written
else
	error reading
(and naturally return at end of function)
```